test/rules/parameter_assignments_test.dart

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(ParameterAssignmentsTest);
+  });
+}
+
+@reflectiveTest
+class ParameterAssignmentsTest extends LintRuleTest {
+  @override
+  String get lintRule => 'parameter_assignments';
+
+  @FailingTest(reason: 'Closures not implemented')
+  test_closure_assignment() async {
+    await assertDiagnostics(r'''
+void f() {
+  (int p) {
+    p = 2;
+  }(2);
+}
+''', [
+      lint(27, 5),
+    ]);
+  }
+
+  test_function_assignment() async {
+    await assertDiagnostics(r'''
+void f(int p) {
+  p = 4;
+}
+''', [
+      lint(18, 5),
+    ]);
+  }
+
+  test_function_incrementAssignment() async {
+    await assertDiagnostics(r'''
+void f(int p) {
+  p += 4;
+}
+''', [
+      lint(18, 6),
+    ]);
+  }
+
+  test_function_named_default() async {
+    await assertDiagnostics(r'''
+void f({int p: 5}) {
+  print(p++);
+}
+''', [
+      lint(29, 3),
+    ]);
+  }
+
+  test_function_named_optional_ok() async {
+    await assertNoDiagnostics(r'''
+void f({int? optional}) {
+  optional ??= 8;
+}
+''');
+  }
+
+  test_function_ok() async {
```

Comment:
nit: rather than 'ok' I'd say why it was ok. In this case perhaps something like 'noAssignment'.